.ci/docs.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+NAME=$0
+COMMAND=$1
+
+function usage {
+    echo "usage: $NAME <command>"
+    echo
+    echo "  install  Install dependencies for building docs"
+    echo "  run      Build documentation"
+    echo "  upload   Upload documentation to gh-pages branch"
+    exit 1
+}
+
+if [[ "$COMMAND" == "install" ]]; then
+    conda install matplotlib scipy
+    pip install jupyter_client sphinx nbsphinx sphinx_rtd_theme ghp-import
```

Comment:
Why are we installing some dependencies with conda and others with pip?
Also, do we have to hide the dependencies in this file? I'd prefer to properly list those in `setup.py` (as optional dependency where appropriate). If that does not work, a requirements file seems better to me (because than we are listing the dependency in two places anyways assuming setup.py lists them too and a requirements file is seems more discoverable).

Reply:
I literally have no idea. @tbekolay would you care to enlighten me and @jgosmann?

Reply:
We use Conda with TravisCI to speed up builds significantly -- at least, that was the main intention before wheels came along. However, even now I believe the wheels situation on Linux is not great, so you're likely to be building NumPy from source if you `pip install numpy`. That takes a long time, and is not likely to give you a very efficient install of NumPy. Conda will give you a binary that uses MKL, so it's fast to install and fast to run.

NumPy is the main reason to use Conda, and then if you're already using Conda, you might as well use it for other packages that need to build C extensions (e.g. scipy) and packages with tricky installations (e.g. Matplotlib, or at least it used to be tricky, I think it's fine now).

You could replace the `pip install` line with a `pip install nengo[docs]` or something, but there are some things (like `ghp-import`) that are specific to TravisCI and are not actually required to build the docs. It's up to the Nengo SPA maintainer how exactly you want to organize these things, but for me, I see setting up TravisCI as a slightly different task than setting up a user's environment. For TravisCI, I want the minimal set of packages installed as quickly as possible to run the build job, and ideally I want the most recent version of those packages so that we can find any issues arising from updates in dependent packages. For a user's environment, I wouldn't want to prescribe that much and would rather keep things generic (e.g. if you want to build the docs, do `pip install nengo[docs]`); if they already have some older version of Sphinx installed, then that's most likely fine, and if not I can recommend upgrading.

Reply:
It turns out you can now force `pip` to install binaries with the option `pip install --only-binary=:all:` and it seems to work well (on my machine). However, since `conda` was already doing the virtual environment management, removing this dependency would mean having to figure out a different way to do this, which might not be worth it...

Reply:
If the tests are no slower using only pip, then it definitely makes sense to switch. I would try it and see. The virtual environment stuff doesn't matter on TravisCI because it is itself an isolated environment (a docker container); we don't have to use virtual environments (we only did in conda because everything is already in a virtual environment, essentially, so might as well).